src/bootstrap/lib.rs

```diff
@@ -769,12 +772,51 @@ impl Build {
     fn release(&self, num: &str) -> String {
         match &self.config.channel[..] {
             "stable" => num.to_string(),
-            "beta" => format!("{}-beta{}", num, channel::CFG_PRERELEASE_VERSION),
+            "beta" => format!("{}-beta.{}", num, self.beta_prerelease_version()),
             "nightly" => format!("{}-nightly", num),
             _ => format!("{}-dev", num),
         }
     }
 
+    fn beta_prerelease_version(&self) -> u32 {
+        if let Some(s) = self.prerelease_version.get() {
+            return s
+        }
+
+        // We're going to be doing some inspection of the git repository. If
+        // we're on CI we typically do a shallow clone so attempt to "unshallow"
+        // the repository and turn it into a full clone.
+        if env::var("CI").is_ok() {
+            self.try_run_quiet(
+                Command::new("git")
+                    .arg("fetch")
+                    .arg("--unshallow"),
```

Comment:
Since the bootstrap is running inside Docker where the repository is mount as read-only, I'm not sure if `git fetch --unshallow` will succeed.

This command may need to be run before the docker is built (before `src/ci/docker/run.sh` finishes), but it should only be run in the beta branch.

Reply:
Gah right good point, what a bummer! I think I may try to fold this into the `init_repo.sh` script perhaps.